core/src/io/anuke/mindustry/world/blocks/power/PowerNode.java

```diff
@@ -299,4 +317,27 @@ protected void drawLaser(Tile tile, Tile target){
         Draw.color();
     }
 
+    public static boolean insulated(Tile tile, Tile other){
+        return insulators(tile, other).size > 0;
+    }
+
+    public static Array<Tile> insulators(Tile tile, Tile other) {
+        Array<Tile> tiles = new Array<Tile>();
```

Comment:
Don't create any new objects in a method like this, that may be called many times per frame. If you absolutely must use an array, use a static temporary one and clear it every time.

Reply:
Eventually i had to go with an atomic boolean to get the result from the lambda 🙊